psetv2/updater.go

```diff
@@ -483,81 +463,81 @@ func (arg AddInReissuanceArgs) areAddressesConfidential() bool {
 // the provided entropy, blinder and amounts and attaches it to the new input.
 // NOTE: This transaction must be blinded later so that a new token blinding
 // nonce is generated for the new token output
-func (u *Updater) AddInReissuance(arg AddInReissuanceArgs) error {
-	if err := arg.validate(); err != nil {
+func (u *Updater) AddInReissuance(inputIndex int, arg AddInReissuanceArgs) error {
+	if err := u.validateInputIndexForIssuance(inputIndex); err != nil {
 		return err
 	}
 
-	if len(u.Pset.Inputs) == 0 {
-		return fmt.Errorf(
-			"transaction must contain at least one input before adding a reissuance",
-		)
+	if err := arg.validate(); err != nil {
+		return err
 	}
 
 	p := u.Pset.Copy()
 
-	// add input
-	if err := p.addInput(arg.TokenPrevOut.toPartialInput()); err != nil {
+	entropy, err := hex.DecodeString(arg.Entropy)
```

Comment:
we should make sure that entropy is correct in args validation function rather than returning an error here. If the entropy is not an hex or its not exactly `32`  bytes then we return an `invalid issuance entropy` error message